src/cmd/build.rs

```diff
@@ -626,4 +637,39 @@ mod tests_ci_only {
             Ok(())
         })
     }
+
+    #[test]
+    fn optimization_passes_from_profile_must_be_used() {
+        with_tmp_dir(|path| {
+            // given
+            cmd::new::execute("new_project", Some(path)).expect("new project creation failed");
+            let cargo_toml_path = path.join("new_project").join("Cargo.toml");
+            write_optimization_passes_into_manifest(&cargo_toml_path, OptimizationPasses::Three);
+            let cmd = BuildCommand {
+                manifest_path: Some(cargo_toml_path),
+                build_artifact: BuildArtifacts::All,
+                verbosity: VerbosityFlags::default(),
+                unstable_options: UnstableOptions::default(),
+
+                // we choose no optimization passes as the "cli" parameter
+                optimization_passes: None,
+            };
+
+            // when
+            let res = cmd.exec().expect("build failed");
+            let optimization = res
+                .optimization_result
+                .expect("no optimization result available");
+
+            // then
+            assert!(
+                optimization.optimized_size < optimization.original_size * 0.5,
+                "The optimized size {:?} DOES NOT differ enough from the original size {:?}",
+                optimization.optimized_size,
+                optimization.original_size
+            );
```

Comment:
This 0.5 magic number looks like a hack tbh ... is there no better solution to testify?